packages/calling/src/Contacts/ContactsClient.ts

```diff
@@ -253,75 +260,59 @@ export class ContactsClient implements IContacts {
-      const {contactType, avatarUrlDomain, encryptionKeyUrl, ownerId, groups} =
-        contactsDataMap[contactId];
-      let avatarURL = '';
-
-      if (photos.length) {
-        avatarURL = photos[0].value;
-      }
-      const addedPhoneNumbers = contactsDataMap[contactId].phoneNumbers;
+    inputList: SCIMListResponse
+  ): Contact[] | null {
+    const finalContactList: Contact[] = [];
 
-      if (addedPhoneNumbers) {
-        const decryptedPhoneNumbers = await this.decryptContactDetail(
-          encryptionKeyUrl,
-          addedPhoneNumbers
-        );
-
-        decryptedPhoneNumbers.forEach((number) => phoneNumbers.push(number));
-      }
-
-      const addedSipAddresses = contactsDataMap[contactId].sipAddresses;
+    try {
+      const contactList = Object.keys(contactsDataMap);
+      for (let n = 0; n < contactList.length; n += 1) {
+        const filteredContact = inputList.Resources.filter((item) => item.id === contactList[n])[0];
+
+        const {displayName, emails, phoneNumbers, photos} = filteredContact;
+        const {sipAddresses} = filteredContact[SCIM_WEBEXIDENTITY_USER];
+        const firstName = filteredContact.name.givenName;
+        const lastName = filteredContact.name.familyName;
+        const manager = filteredContact[SCIM_ENTERPRISE_USER].manager.displayName;
+        const department = filteredContact[SCIM_ENTERPRISE_USER].department;
+
+        let avatarURL = '';
+        if (photos?.length) {
+          avatarURL = photos[0].value;
+        }
 
-      if (addedSipAddresses) {
-        const decryptedSipAddresses = await this.decryptContactDetail(
+        const {contactType, avatarUrlDomain, encryptionKeyUrl, ownerId, groups} =
+          contactsDataMap[contactList[n]];
+
+        const cloudContact = {
+          avatarUrlDomain,
+          avatarURL,
+          contactId: contactList[n],
+          contactType,
+          department,
+          displayName,
+          emails,
           encryptionKeyUrl,
-          addedSipAddresses
-        );
-
-        decryptedSipAddresses.forEach((address) => sipAddresses.push(address));
+          firstName,
+          groups,
+          lastName,
+          manager,
+          ownerId,
+          phoneNumbers,
+          sipAddresses,
+        };
+
+        finalContactList.push(cloudContact);
       }
-
-      const cloudContact = {
-        avatarUrlDomain,
-        avatarURL,
-        contactId,
-        contactType,
-        department,
-        displayName,
-        emails,
-        encryptionKeyUrl,
-        firstName,
-        groups,
-        lastName,
-        manager,
-        ownerId,
-        phoneNumbers,
-        sipAddresses,
-        title: jobTitle,
-      };
-
-      return cloudContact;
     } catch (error: any) {
       Logger.error(new ExtendedError(error.message, {}, ERROR_TYPE.DEFAULT), {});
 
       return null;
     }
+
+    return finalContactList;
```

Comment:
Would it be better to put this in a .finally block

Reply:
@Shreyas281299 are there any advantages to doing this?